packages/time/src/core/calendar.ts

```diff
@@ -0,0 +1,397 @@
+    const firstDayOfMonth = this.getFirstDayOfMonth()
+    const firstDayOfWeek = this.getFirstDayOfWeek()
+
+    switch (this.store.state.viewMode.unit) {
+      case 'month': {
+        const firstDayOfPrevMonth = firstDayOfMonth.subtract({
+          months: this.store.state.viewMode.value,
+        })
+        this.store.setState((prev) => ({
+          ...prev,
+          currentPeriod: firstDayOfPrevMonth,
+        }))
+        break
+      }
+
+      case 'week': {
+        const firstDayOfPrevWeek = firstDayOfWeek.subtract({
+          weeks: this.store.state.viewMode.value,
+        })
+        this.store.setState((prev) => ({
+          ...prev,
+          currentPeriod: firstDayOfPrevWeek,
+        }))
+        break
+      }
+
+      case 'day': {
+        const prevCustomStart = this.store.state.currentPeriod.subtract({
+          days: this.store.state.viewMode.value,
+        })
+        this.store.setState((prev) => ({
+          ...prev,
+          currentPeriod: prevCustomStart,
+        }))
+        break
+      }
+    }
+  }
+
+  goToNextPeriod() {
+    const firstDayOfMonth = this.getFirstDayOfMonth()
+    const firstDayOfWeek = this.getFirstDayOfWeek()
+
+    switch (this.store.state.viewMode.unit) {
+      case 'month': {
+        const firstDayOfNextMonth = firstDayOfMonth.add({
+          months: this.store.state.viewMode.value,
+        })
+        this.store.setState((prev) => ({
+          ...prev,
+          currentPeriod: firstDayOfNextMonth,
+        }))
+        break
+      }
+
+      case 'week': {
+        const firstDayOfNextWeek = firstDayOfWeek.add({
+          weeks: this.store.state.viewMode.value,
+        })
+        this.store.setState((prev) => ({
+          ...prev,
+          currentPeriod: firstDayOfNextWeek,
+        }))
+        break
+      }
+
+      case 'day': {
+        const nextCustomStart = this.store.state.currentPeriod.add({
+          days: this.store.state.viewMode.value,
+        })
+        this.store.setState((prev) => ({
+          ...prev,
+          currentPeriod: nextCustomStart,
+        }))
+        break
+      }
+    }
+  }
+
+  goToCurrentPeriod() {
+    this.store.setState((prev) => ({
+      ...prev,
+      currentPeriod: Temporal.Now.plainDateISO(),
+    }))
+  }
+
+  goToSpecificPeriod(date: Temporal.PlainDate) {
+    this.store.setState((prev) => ({
+      ...prev,
+      currentPeriod: date,
+    }))
+  }
+
+  updateCurrentTime() {
+    this.store.setState((prev) => ({
+      ...prev,
+      currentTime: Temporal.Now.plainDateTimeISO(),
+    }))
+  }
+
+  getEventProps(id: Event['id']) {
+    return getEventProps(this.getEventMap(), id, this.store.state)
+  }
+
+  getCurrentTimeMarkerProps(): {
+    style: CSSProperties
+    currentTime: string | undefined
+  } {
+    const { hour, minute } = this.store.state.currentTime
+    const currentTimeInMinutes = hour * 60 + minute
+    const percentageOfDay = (currentTimeInMinutes / (24 * 60)) * 100
+
+    return {
```

Comment:
This is also layout specific, so sort of depends on how the developer renders their calendar